### ISSUE TITLE
Difficulties adding Billing plugin

### ISSUE THREAD

**Issue by HarrisonShort** [google/play-unity-plugins#45]
Hey there! This is actually an issue centric to the Google Play billing plugin, however I cannot seem to find an appropriate place to get in touch about any issues we may be having, and this appears to be the best possible option. Please let me know if there's a better place for this and I'll be sure to enquire there!

Currently my team is trying to add the Google Play billing plugin to our project. The main reason for this is that we want to utilize the ability to specify proration mode when upgrading and downgrading subscriptions, which is not currently offered in UnityIAP. We are running into issues adding the plugin to our existing project. 

I am following the instructions at [this page](https://developer.android.com/google/play/billing/unity).

**Issue 1:**
When getting to step 3 of "For Unity 2018.4 and above" instructions, the prompt for "Add Selected Registries" does not actually appear. 

**Issue 2:** 
The note in regards to the External Dependency Manager (which I'm assuming is the Game Package Registry for Unity - it's a little confusing that the terminology differs here) is slightly ambiguous. We actually do have the Package Manager version of the EDM, and following those steps shows two packages (google-signin-plugin and play-services-resolver) but the tutorial doesn't really elaborate on what you should be doing when reaching this popup. 

The fact that we already have the External Dependency Manager may explain Issue 1 above?

**Issue 3:**
This is the primary issue, I believe. Once installing the Google Play Billing package from the Package Manager, a number of errors appear in the Unity console.
![image](https://user-images.githubusercontent.com/18225638/91937295-f1db8800-ed34-11ea-9d23-cbf7ee448c20.png)
There are 34 of these in total at the bottom of the console, as well as the following that appears at the very top of the console: 
`Error: Could not load signature of UnityEditor.Purchasing.GooglePlayProductCatalogExporter:ProductTypeString due to: Could not load file or assembly 'UnityEngine.Purchasing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. assembly:UnityEngine.Purchasing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null type:<unknown type> member:(null) signature:<none>`
I can actually get past all these error messages (_except_ the pasted error) by restarting Unity, but when I attempt to call anything from the Google.Play.Billing namespace, the only options are:
- Editor (and deeper options)
- GooglePlayStoreProrationMode
- GooglePlayStoreUnsupportedException
- Internal (and deeper options)

---

**What I've attempted 1:**
The given installation steps [here](https://developer.android.com/google/play/billing/unity) as closely as possible, with the above issues.

**What I've attempted 2:** 
Removing the Package Manager version of External Dependency Manager, along with the dependent Firebase packages, and then attempting the given installation steps as closely as possible again.

**What I've attempted 3:**
The "For Unity 5.6 to 2018.3" installation steps [here](https://developer.android.com/google/play/billing/unity). Each way I've attempted to add the plugin has resulted in the same issues.

Also might be worth noting that we've been using UnityIAP successfully in our live app for about 10 months now, so those UnityIAP steps at the beginning of the installation steps were not completed, but after reviewing them I believe they were all done prior anyway.

---

I do get the feeling that this is a result of user error rather than something within the plugin/installation process, especially since this sort of thing isn't my forte, so any guidance to fixing this would be much appreciated! Hope to hear from one of you folks soon!

Unity - 2019.2.17f (MacOS)
External Dependency Manager - 1.2.156 (Package Manager) / 1.2.157 (in Assets)
Google Play Billing Library - 3.0.2
In App Purchasing - 2.1.0
UnityPurchasing - 1.23.5

**Comment by Qing451800** [comment]
For issue 1, it should be just a transient issue and restart Unity will help.
For issue 2, @thanwah can comment.
For issue 3, the compilation errors are complaining about the classes defined in Unity IAP. Please make sure you do turn on and install the Unity IAP. The process is a little bit tricky because if you have compilation error then you cannot install Unity IAP correctly. I'd suggest you first set up Unity IAP, then import Play Billing Plugin.
As of the class in Google.Play.Billing namespace, you should be able to have access to these classes:
Google.Play.Billing.GooglePlayStoreModule
Google.Play.Billing.IGooglePlayConfiguration
Google.Play.Billing.IGooglePlayStoreExtensions

**Comment by HarrisonShort** [comment]
> For issue 1, it should be just a transient issue and restart Unity will help.
I have restarted Unity a few times, and unfortunately this agreement still hasn't appeared, so I'm not entirely sure what's happening there. I'll keep trying though!

> For issue 3, the compilation errors are complaining about the classes defined in Unity IAP. Please make sure you do turn on and install the Unity IAP. The process is a little bit tricky because if you have compilation error then you cannot install Unity IAP correctly. I'd suggest you first set up Unity IAP, then import Play Billing Plugin.
We've been using UnityIAP for a while now, without any troubles. I'll attempt to remove it from our project, and then begin the process again from the start to see if there is any progress. Thanks for your response!

**Comment by HarrisonShort** [comment]
I've now completed the following steps, and the plugin appears to be working correctly. I can now access the full range of classes that should be available. If you are having similar troubles, try this out:

1. Removed UnityIAP (Assets/Plugins/UnityPurchasing and Package Manager In App Purchasing)
2. Disabled In-App Purchasing from Services menu
3. Removed any references to UnityIAP classes from the project (to remove type/namespace errors)
4. Re-enabled In-App Purchasing from Services menu, which reimports UnityPurchasing
5. Re-added In App Purchasing from Package Manager
6. Installed Google Play Billing from Package Manager
7. (Optional, I think) Remove Assets/External Dependency Manager, since we have the Package Manager version already installed for Firebase packages

I'll close the issue now, but I do think that some of my other queries are still valid, the installation process is not 100% fleshed out at this stage and doesn't cover a lot of the possible edge cases that may occur. 

@Qing451800 @thanwah Appreciate your hard work!
